src/server/extractors.rs

```diff
@@ -0,0 +1,35 @@
+use regex::Regex;
+use actix_web::{
+    Error, FromRequest,
+};
+use futures::future::{ok, err, Ready};
+use actix_web::error::ErrorBadRequest;
+
+pub struct  ClientState {
+   extractedClientState: String,
+   validatedClientState: String,
+}
+
+impl FromRequest for ClientState {
+    type Config = ();
+    type Error = Error;
+    type Future = Ready<Result<Self, Self::Error>>;
+
+    fn from_request(req: &::actix_web::HttpRequest,
+                    payload: &mut actix_web::dev::Payload,) -> Self::Future {
+
+        let client_state: &str = req.headers().get("X-Client-State").unwrap()
```

Comment:
good for now, but we probably want to call `unwrap_or_else` on these so that we capture the error correctly.